c/datasetjson.c

```diff
@@ -900,11 +904,11 @@ static void respondWithDYNALLOCError(HttpResponse *response,
 #define IS_DAMEMBER_EMPTY($member) \
   (!memcmp(&($member), &(DynallocMemberName){"        "}, sizeof($member)))
 
-static void updateDatasetWithJSONInternal(HttpResponse* response,
+static int updateDatasetWithJSONInternal(HttpResponse* response,
```

Comment:
it looks like the return `int` is only ever `-1`, so you may want to add some different ints for different error reasons, like `-2`, etc... or more commonly, you could do:

0=ok
1=problem
2= more problems
3....
Or,

0=ok
4=warn
8=severe